service/src/main/java/org/exoplatform/termsconditions/TermsAndConditionsService.java

```diff
@@ -0,0 +1,90 @@
+package org.exoplatform.termsconditions;
+
+import org.exoplatform.rest.response.TermsAndConditions;
+import org.exoplatform.service.FunctionalConfigurationService;
+import org.exoplatform.service.exception.FunctionalConfigurationRuntimeException;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.OrganizationService;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.utils.NodeUtils;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.version.Version;
+
+import java.util.Objects;
+
+import static org.exoplatform.utils.IdentityUtils.findUserProfileByUserName;
+
+public class TermsAndConditionsService {
+
+    private static final String TERMS_AND_CONDITONS_PROPERTY = "acceptedTermsAndConditions";
+    private static final Log LOGGER = ExoLogger.getLogger(TermsAndConditionsService.class);
+
+    private final FunctionalConfigurationService functionalConfigurationService;
+
+    private IdentityManager identityManager;
+
+    public TermsAndConditionsService(FunctionalConfigurationService functionalConfigurationService, IdentityManager identityManager) {
+        this.functionalConfigurationService = functionalConfigurationService;
+        this.identityManager = identityManager;
+    }
+
+    boolean isTermsAndConditionsAcceptedBy(String userName){
+        Profile socialProfile = findUserProfileByUserName(userName);
+
+        try {
+            String acceptedTermsVersion = socialProfile.getProperty(TERMS_AND_CONDITONS_PROPERTY) + "";
+            String currentVersionUid = getCurrentTermsAndConditionsVersionUUID();
+
+            return currentVersionUid.equals(acceptedTermsVersion);
+        } catch (Exception e) {
+            LOGGER.warn("Terms and conditions still not accepted by user: " + userName, e);
+            return false;
+        }
+    }
+
+    private String getCurrentTermsAndConditionsVersionUUID() {
+
+        TermsAndConditions termsAndConditions = functionalConfigurationService.getTermsAndConditions();
+        Node termsAndConditionsNode = NodeUtils.findCollaborationFile(termsAndConditions.getWebContentUrl());
+
+        try {
+            Version baseVersion = termsAndConditionsNode.getBaseVersion();
+            return baseVersion.getUUID();
```

Comment:
I think it should be better to store the version number instead of the node uuid, it would make it easier to troubleshoot.

Reply:
I am using the uuid because if we change the document with the same version as the precedent we cannot detect the changment. With UUID, we are validating the document and the version number.